cherrypy/lib/cptools.py

```diff
@@ -630,3 +630,21 @@ def set_response_header():
         v.sort()
         resp_h['Vary'] = ', '.join(v)
     request.hooks.attach('before_finalize', set_response_header, 95)
+
+
+def convert_params(exception, error=400):
+    """Convert request params based on function annotations, with error handling.
+
+    exception
+        Exception class to catch.
```

Comment:
When would the exception class ever not be a TypeError? If there aren't such cases, perhaps this detail could be omitted. If there are such cases, perhaps it would be suitable to supply `TypeError` as the default for `exception` as it's likely the most common usage.